api/ccapi/include/optimizer.h

```diff
@@ -83,8 +83,8 @@ class Optimizer {
    *
    * @param lrs the learning rate scheduler object
    */
-  virtual void setLearningRateScheduler(
-    std::unique_ptr<ml::train::LearningRateScheduler> &&lrs) = 0;
+  virtual int setLearningRateScheduler(
+    std::shared_ptr<ml::train::LearningRateScheduler> lrs) = 0;
```

Comment:
is there any reason you change the unique_ptr to shared_ptr?

Reply:
Discussed offilne